### ISSUE TITLE
Cache location/icon not updated by nearby search result

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#5902]
##### Detailed steps causing the problem:
- Premium member
- New map
- Open live map
- Scroll to new position to trigger loading and wait until fully loaded
- Note that all caches have orange circles although acc. to log a nearby search has been done in addition to live map loading
- Click on a cache next to map center
##### Actual behavior after performing these steps:

Cache jumps to correct position and orange circle vanishes instantly wirhout further network activity.
So the data from nearby search is there but probably the map is not updated
##### Expected behavior after performing these steps:

Cache positions to be updated after nearby search was completed in the background.
This is tested and working in the same scenario with the "old map".
##### Version of c:geo used:

2016.08.14
##### Is the problem reproducible for you?

Yes

**Comment by Lineflyer** [comment]
Together or at least maybe linked with this issue people report, that caches vanish and reappear on the new map when moving the map or even while the map is loaded.

@rsudev Just an an indication. Now, that this version of the new map is out this issue is one of the most mentioned on support mail. I set the label to "High" to represent this.

**Comment by SammysHP** [comment]
I noticed this issue just this weekend with the latest nightly, but never noticed it before (last update several weeks ago).

**Comment by Lineflyer** [comment]
It was there before and it started IMHO with a fix for another issue. Let me try to look up where I think it comes from.

**Comment by Lineflyer** [comment]
It could either be since https://github.com/cgeo/cgeo/commit/faf9cbe549e0c766c45ded0db6462f6ef8424c6b or since https://github.com/cgeo/cgeo/commit/9f30101c1467a6a3b3095c17fa86afeb4cd8b780

**Comment by jonas-koeritz** [comment]
> Together or at least maybe linked with this issue people report, that caches vanish and reappear on the new map when moving the map or even while the map is loaded.

Just for reference: This hasn't happened to me since I'm running with the new Map API (#7514).

**Comment by rsudev** [comment]
This is likely an accidental side-effect of the better performance of the new gc.com web api. Try it over GPRS and you likely experience this problem again.
I am working on rectifying the underlying wrong design in my new map implementation.

**Comment by jonas-koeritz** [comment]
You should have a look at #7337 too, I had a quick look at it and the calculation of the map dimensions for low zoom levels (0 or 1, also occuring on initialization of the live map) seem to be the cause.

**Comment by Lineflyer** [comment]
Obsolete after #7514 was merged as we do no onger not use the nearby search in parallel to map loading.
